src/utils.rs

```diff
@@ -14,3 +21,65 @@ where
     U: Send + 'static,
 {
 }
+
+pub fn derive_gas_estimation_overhead(
+    gas_limit: u32,
+    gas_price_per_pubdata: u32,
+    encoded_len: usize,
+) -> u32 {
+    // Even if the gas limit is greater than the MAX_TX_ERGS_LIMIT, we assume that everything beyond MAX_TX_ERGS_LIMIT
+    // will be spent entirely on publishing bytecodes and so we derive the overhead solely based on the capped value
+    let gas_limit = std::cmp::min(MAX_TX_ERGS_LIMIT, gas_limit);
+
+    // Using large U256 type to avoid overflow
+    let max_block_overhead = U256::from(block_overhead_gas(gas_price_per_pubdata));
+    let gas_limit = U256::from(gas_limit);
+    let encoded_len = U256::from(encoded_len);
+
+    // The MAX_TX_ERGS_LIMIT is formed in a way that may fullfills a single-instance circuits
+    // if used in full. That is, within MAX_TX_ERGS_LIMIT it is possible to fully saturate all the single-instance
+    // circuits.
+    let overhead_for_single_instance_circuits =
+        ceil_div_u256(gas_limit * max_block_overhead, MAX_TX_ERGS_LIMIT.into());
+
+    // The overhead for occupying the bootloader memory
+    let overhead_for_length = ceil_div_u256(
+        encoded_len * max_block_overhead,
+        BOOTLOADER_TX_ENCODING_SPACE.into(),
+    );
+
+    // The overhead for occupying a single tx slot
+    let tx_slot_overhead = ceil_div_u256(max_block_overhead, MAX_TXS_IN_BLOCK.into());
+
+    vec![
+        (0.1 * overhead_for_single_instance_circuits.as_u32() as f64).floor() as u32,
+        overhead_for_length.as_u32(),
+        tx_slot_overhead.as_u32(),
+    ]
+    .into_iter()
+    .max()
+    .unwrap()
```

Comment:
Please avoid using `unwrap()` where possible.

Reply:
I can update to `unwrap_or(0)`, but this code is copy/pasted from production and would prefer to leave untouched. As the above vector is guaranteed to have these 3 values in them, "max" will always return a value.

Reply:
Updated to `.unwrap_or(0)`